File: features/rolemanagement/src/main/java/com/nivasafinance/features/rolemanagement/mapping/repository/RolePermissionMappingRepository.java
package com.nivasafinance.features.rolemanagement.mapping.repository;

import com.nivasafinance.features.rolemanagement.mapping.entity.RolePermissionMapping;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import java.util.List;

@Repository
public interface RolePermissionMappingRepository extends JpaRepository<RolePermissionMapping, Long> {
    List<RolePermissionMapping> findByRoleIn(List<String> roles);
}

File: features/rolemanagement/src/main/java/com/nivasafinance/features/rolemanagement/mapping/repository/RolePermissionMappingRepository.kt
package com.nivasafinance.features.rolemanagement.mapping.repository

import com.nivasafinance.features.rolemanagement.mapping.entity.RolePermissionMapping
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository
import java.util.UUID

@Repository
interface RolePermissionMappingRepository : JpaRepository<RolePermissionMapping, UUID>
{
    fun findByRoleIdIn(roleIds: List<UUID>): List<RolePermissionMapping>
}
